### PR TITLE
Add new QL endpoints of /v2/attr and /v2/attr/{attr_name}

### PR DESCRIPTION
**Pull request by tzuchen-liu**

### CODE REVIEW

**Review by tstorek — CHANGES_REQUESTED** [review]
@orange95 thanks, for this pull request. Unfortunatly, there are no tests provided for the functionality. Please, do so. Additionally, it would great if you state why you deviate from the TimeSeries pattern. I think, that all queries that actually contain values should be converted to Either TimeSeries objects or to list of TimeSeriesObjects

**Comment by orange95** [comment]
as for adding test, I plan to add these to the line 126 of tests/clients/test_ngsi_v2_timeseries.py
now need to pull in order to commit to another issue, so I put the code here

`            entity_attr = client.get_entity_by_attrs(
                entity_type=self.entity_1.type)
            for entity in entity_attr:
                print(entity.to_pandas())

            entity_attr_name = client.get_entity_by_attr_name(
                attr_name="temperature", entity_type=self.entity_1.type)
            for entity in entity_attr_name:
                print(entity.to_pandas())


`

**Comment by tstorek** [comment]
@SBlechmann is there still any progress on this?

**Comment by SBlechmann** [comment]
Yes, we are still on it, sorry for the delay.

**Comment by SystemsPurge** [comment]
@tstorek First , appologies for the long list of commits, only the last three are relevant, the rest stems from a rebase on master because the branch was old. The methods now return the sought after timeseries/timeseries list object. I could not find a way to avoid nested loops, given that the response itself comprises of nested/categorized elements. 
One important detail : The get_entity_by_attrs method does not combine the timeseries of same entities for different attributes, since extending the values/index for different attributes doesnt seem to be a trivial task : If done properly, the timestamps in the indexes should be compared one by one, to determine whether adding a None value for a specific attribute is necessary. 
Hence why in the test, as u can see, the method now returns a timeseries list of size : number of unique entities x number of unique attributes. The question then becomes, is it ok for this method to return such a result OR, should a timeseries method be implemented, that combines timeseries of same entities with different attributes?

**Review by djs0109 — CHANGES_REQUESTED** [review]

File: filip/clients/ngsi_v2/quantumleap.py [hunk 1]
```diff
@@ -1074,43 +1063,148 @@ def get_entity_attr_values_by_type(self,
...
+                                   from_date=from_date,
+                                   to_date=to_date,
+                                   last_n=last_n,
+                                   limit=limit,
+                                   offset=offset,
+                                   georel=georel,
+                                   geometry=geometry,
+                                   coords=coords,
+                                   aggr_scope=aggr_scope)
+        return [TimeSeries(index=item.get('index'),
```
[the remark this answers is not in the document]
Comment (SystemsPurge) [reply filip/clients/ngsi_v2/quantumleap.py:1083]: Could you please give me more information on how to reproduce this ? What versions of the GEs are you running ? Did this error come up in the unittests ?
 I've just re-ran the tests ( with added types to account for that scenario ) and it still works for me.
grafik

grafik

**Review by djs0109 — APPROVED** [review]

**Review by djs0109 — APPROVED** [review]

**Pull request merged**